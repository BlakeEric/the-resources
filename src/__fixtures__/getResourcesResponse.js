const getResourcesResponse = [
  {
    node: {
      id: 'd04da249-50e7-5b18-a992-d530e14831a7',
      frontmatter: {
        title: 'A List Apart',
        format: 'Article',
        focus: 'Front-end',
        skillLevel: ['Beginner','Moderate','Advanced']
      },
      fields: {
        slug: '/a-list-apart/'
      }
    }
  },
  {
    node: {
      id: '77a6be63-d3c7-50cb-bd29-438427560388',
      frontmatter: {
        title: 'Scrimba',
        format: 'Interactive',
        focus: 'Front-end',
        skillLevel: ['Beginner','Moderate']
      },
      fields: {
        slug: '/scrimba/'
      }
    }
  },
  {
    node: {
      id: '28feb8c5-acb2-5638-bdd6-b0457eff8db3',
      frontmatter: {
        title: 'Smashing Magazine',
        format: 'Article',
        focus: 'Multiple',
        skillLevel: ['Beginner','Moderate','Advanced']
      },
      fields: {
        slug: '/smashing-magazine/'
      }
    }
  },
  {
    node: {
      id: '744d8211-8c35-5c83-b8c1-ecb4ab48c0db',
      frontmatter: {
        title: 'The Odin Project',
        format: 'Tutorial',
        focus: 'Multiple',
        skillLevel: ['Beginner','Moderate']
      },
      fields: {
        slug: '/the-odin-project/'
      }
    }
  },
  {
    node: {
      id: '3b19247d-2ddd-558d-b0ce-14ec1972d52f',
      frontmatter: {
        title: 'Watch and Code',
        format: 'Video',
        focus: 'Front-end',
        skillLevel: ['Beginner']
      },
      fields: {
        slug: '/watch-and-code/'
      }
    }
  }
]

export default getResourcesResponse;
