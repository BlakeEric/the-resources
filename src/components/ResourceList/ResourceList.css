.table {
    border-collapse: collapse;
}
.table tbody tr {
    border-bottom: 1px solid #eaeaea; 
}