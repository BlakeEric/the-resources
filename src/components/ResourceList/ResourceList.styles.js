import { css } from '@emotion/core'
//import { rhythm } from "../../utils/typography"

export const resourceListStyles = css`
overflow-x: scroll;
border-radius: 5px;
padding: 1rem;
boxShadow: 0 2px 2px 0 rgba(0,0,0,0.05);
border: 1px solid #eaeaea;
table {
  min-width: 600px;
  position: relative;
}
tbody > tr {
  border-bottom: 1px solid #eaeaea;
}
td {
  width: 25%;
  padding-top: 0.25rem;
}
td span.notFoundMessage {
  width: 100%;
  text-align: center;
  display: inline-block;
}
td a {
  display: inline-block;
  line-height: 1.25;
}
`