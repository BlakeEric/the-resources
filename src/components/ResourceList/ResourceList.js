import React, { useContext } from "react"
import { resourceListStyles } from './ResourceList.styles.js'
import "./ResourceList.css"
import ResourceListHeader from '../ResourceListHeader/ResourceListHeader'
import Resource from '../Resource/Resource'
import { ResourceContext } from '../ResourceContext/ResourceContext'


export default function ResourceList() {

  const context = useContext(ResourceContext);
  const { resources } = context;
  const filteredResources = context.getFilteredResources();

  if (resources.length < 1) {
    return (
      <section id="resources">
        <h4>Something went wrong :(</h4>
        <p>We are unable to retrieve resources at this time.</p>
      </section>
    )
  }

  return (
    <div>
      <section className="tableWrap" css={resourceListStyles}>
        <table className="table">
          <ResourceListHeader keys={['name', 'format', 'focus', 'skillLevel']} />
          <tbody>
            {filteredResources.length === 0 ?
              <tr>
                <td colSpan="4">
                  <span className="notFoundMessage">Nothing found matching your search.</span>
                </td>
              </tr>
            : filteredResources.map(({ node }, i) => (
              <Resource
                key={node.id}
                {...node}
                showDetailsAbove={i > filteredResources.length - 5}
              />
            ))}
          </tbody>
        </table>
      </section>
    </div>
  )

}


ResourceList.defaultProps = {
  resources: [],
  skillLevels: [],
  formats:[],
  focuses: []
}

