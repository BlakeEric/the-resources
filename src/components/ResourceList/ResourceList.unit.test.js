import React from "react"
import renderer from "react-test-renderer"
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import getResourcesResponse from '../../__fixtures__/getResourcesResponse'

import ResourceList from "./ResourceList"

const mockProps = {
  resources: getResourcesResponse,
  skillLevels: ["Beginner", "Moderate", "Advanced"],
  focuses: ["Front-end", "Multiple", "Algorithms"],
  formats: ['Article', 'Interactive', 'Tutorial', 'Video']
}

describe("ResourceList", () => {

  /*
  * Make sure the component renders without error
  */
  describe('Initial render', () => {

    it("renders correctly when no resources passed in", () => {
      const tree = renderer
        .create(<ResourceList />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("renders correctly when resources passed in", () => {
      const tree = renderer
        .create(<ResourceList {...mockProps} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("orders ASC by name on initial render", () => {
      const wrapper = mount(<ResourceList {...mockProps} />)

      expect(wrapper.find('tbody tr').first().text().includes('Alligator')).toBeTruthy();
    });
  });

})
