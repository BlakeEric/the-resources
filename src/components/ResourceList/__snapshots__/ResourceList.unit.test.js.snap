// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceList Initial render renders correctly when no resources passed in 1`] = `
<div>
  <section
    className="tableWrap"
    css={
      Object {
        "map": undefined,
        "name": "o8h0qv",
        "next": undefined,
        "styles": "
overflow-x: scroll;
border-radius: 5px;
padding: 1rem;
boxShadow: 0 2px 2px 0 rgba(0,0,0,0.05);
border: 1px solid #eaeaea;
table {
  min-width: 600px;
  position: relative;
}
tbody > tr {
  border-bottom: 1px solid #eaeaea;
}
td {
  width: 25%;
  padding-top: 0.25rem;
}
td span.notFoundMessage {
  width: 100%;
  text-align: center;
  display: inline-block;
}
td a {
  display: inline-block;
  line-height: 1.25;
}
",
        "toString": [Function],
      }
    }
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <button
              className="css-vhv7r1"
              onClick={[Function]}
            >
              Name
               
              â†“
            </button>
          </th>
          <th
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <button
              className="css-vhv7r1"
              onClick={[Function]}
            >
              Format
               
              
            </button>
          </th>
          <th
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <button
              className="css-vhv7r1"
              onClick={[Function]}
            >
              Focus
               
              
            </button>
          </th>
          <th
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <button
              className="css-vhv7r1"
              onClick={[Function]}
            >
              Skill Level
               
              
            </button>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              className="resourceDetails-wrapper"
              css={
                Object {
                  "map": undefined,
                  "name": "1f9ea78",
                  "next": undefined,
                  "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
                  "toString": [Function],
                }
              }
            >
              <a
                onClick={[Function]}
                rel="noreferrer noopener"
                target="_blank"
              >
                Alligator
              </a>
            </div>
          </td>
          <td>
            Article
          </td>
          <td>
            Front-end
          </td>
          <td>
            All
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="resourceDetails-wrapper"
              css={
                Object {
                  "map": undefined,
                  "name": "1f9ea78",
                  "next": undefined,
                  "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
                  "toString": [Function],
                }
              }
            >
              <a
                onClick={[Function]}
                rel="noreferrer noopener"
                target="_blank"
              >
                Badger
              </a>
            </div>
          </td>
          <td>
            Interactive
          </td>
          <td>
            Algorithms
          </td>
          <td>
            Moderate
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="resourceDetails-wrapper"
              css={
                Object {
                  "map": undefined,
                  "name": "1f9ea78",
                  "next": undefined,
                  "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
                  "toString": [Function],
                }
              }
            >
              <a
                onClick={[Function]}
                rel="noreferrer noopener"
                target="_blank"
              >
                Caterpillar
              </a>
            </div>
          </td>
          <td>
            Article
          </td>
          <td>
            Multiple
          </td>
          <td>
            Advanced
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="resourceDetails-wrapper"
              css={
                Object {
                  "map": undefined,
                  "name": "1f9ea78",
                  "next": undefined,
                  "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
                  "toString": [Function],
                }
              }
            >
              <a
                onClick={[Function]}
                rel="noreferrer noopener"
                target="_blank"
              >
                Dolphin
              </a>
            </div>
          </td>
          <td>
            Tutorial
          </td>
          <td>
            Multiple
          </td>
          <td>
            Beginner, Moderate
          </td>
        </tr>
        <tr>
          <td>
            <div
              className="resourceDetails-wrapper"
              css={
                Object {
                  "map": undefined,
                  "name": "1f9ea78",
                  "next": undefined,
                  "styles": "
  position: relative;
  .resourceDetails {
    position: absolute;
    left: 0;
    max-width: 90vw;
    width: 300px;
    top: 110%;
    left: -0.4025rem;
    background: white;
    border-radius: 5px;
    box-shadow: 2px 2px 12px 1px rgba(0,0,0,0.25);
    padding: 0.805rem;
    z-index: 1000;
    border: 1px solid #e4e4e4;
    &:after {
      content: \\"\\";
      height: 12px;
      width: 12px;
      position: absolute;
      top: 0;
      background: white;
      transform: rotate(-45deg);
      transform-origin: 0 0;
      left: 0.805rem;
      border-right: 1px solid #e4e4e4;
      border-top: 1px solid #e4e4e4;
    }
    &.up {
      top: auto;
      bottom: 110%;
      &:after {
        top: auto;
        bottom: 0;
        transform: rotate(45deg);
        transform-origin: 0 100%;
        border-right: 1px solid #e4e4e4;
        border-bottom: 1px solid #e4e4e4;
        border-top: 0;
      }
    }
    h4 {
      margin: 0;
    }
    p {
      color: black;
    }
    p:last-child {
      margin-bottom: 0.805rem;
    }
    .btn-view {
      padding-right: 0.4025rem;
    }
    button, span {
      display: inline-block;
      font-size: 13px;
    }
    .right {
        float: right;
    }
  }
",
                  "toString": [Function],
                }
              }
            >
              <a
                onClick={[Function]}
                rel="noreferrer noopener"
                target="_blank"
              >
                Elephant
              </a>
            </div>
          </td>
          <td>
            Video
          </td>
          <td>
            Front-end
          </td>
          <td>
            Beginner
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;
